public/pages/CreateIndex/components/IndexDetail/IndexDetail.tsx

```diff
@@ -0,0 +1,548 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Ref, forwardRef, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState } from "react";
```

Comment:
Not blocking for this PR, but please break down this component into smaller files. Also try moving interfaces, constants to separate files

Reply:
You are right, the file has over 500 lines, will refractor that as a TODO.